packages/squarev2-integration/src/SquareV2PaymentMethod.tsx

```diff
@@ -0,0 +1,199 @@
+                borderWidth: borderWidthIsFocus,
+                ...inputIsFocus
+            } = styles.focus.input;
+            const {
+                borderColor: borderColorIsError,
+                borderWidth: borderWidthIsError,
+                color: colorIsError,
+            } = styles.error.input;
+
+            return {
+                input,
+                'input.is-focus': {
+                    ...inputIsFocus,
+                },
+                'input.is-error': {
+                    color: colorIsError,
+                },
+                '.input-container': {
+                    borderColor,
+                    borderRadius,
+                    borderWidth,
+                },
+                '.input-container.is-focus': {
+                    borderColor: borderColorIsFocus,
+                    borderWidth: borderWidthIsFocus,
+                },
+                '.input-container.is-error': {
+                    borderColor: borderColorIsError,
+                    borderWidth: borderWidthIsError,
+                },
+                '.message-text': {
+                    color: styles.default.message.color,
+                },
+                '.message-icon': {
+                    color: styles.default.message.color,
+                },
+                '.message-text.is-error': {
+                    color: styles.error.message.color,
+                },
+                '.message-icon.is-error': {
+                    color: styles.error.message.color,
+                },
+            };
+        },
+        [],
+    );
+
+    const [disabled, setDisabled] = useState(true);
+    const onValidationChange = useCallback(
+        (isValid: boolean) => setDisabled(!isValid),
+        [setDisabled],
+    );
+
+    useEffect(() => paymentForm.disableSubmit(method, disabled), [disabled, method, paymentForm]);
+
+    const containerId = 'squarev2_payment_element_container';
+    const initializePayment = useCallback(async () => {
+        let style;
+
+        try {
+            style = mapToSquareStyles(copyStylesFromDummyFormFields());
+        } catch {
+            /* Do nothing: we should not block shoppers from buying. */
+        }
+
+        await checkoutService.initializePayment({
+            gatewayId: method.gateway,
+            methodId: method.id,
+            squarev2: {
+                containerId,
+                style,
+                onValidationChange,
+            },
+        });
+    }, [
+        checkoutService,
+        copyStylesFromDummyFormFields,
+        mapToSquareStyles,
+        method.gateway,
+        method.id,
+        onValidationChange,
+    ]);
+    const deinitializePayment = useCallback(async () => {
+        await checkoutService.deinitializePayment({
+            gatewayId: method.gateway,
+            methodId: method.id,
+        });
+    }, [checkoutService, method.gateway, method.id]);
+
+    useEffect(() => {
```

Comment:
Can we move all the useEffect calls to top of the component? Also maybe move them into a custom hook?

Reply:
This one is not possible due to the `no-use-before-define` rule.

Reply:
Which variable does it complain regarding the no use before define?